ttnn/cpp/ttnn/operations/moreh/moreh_linear_backward/device/moreh_linear_backward_device_operation.cpp

```diff
@@ -0,0 +1,83 @@
+#include "ttnn/tensor/tensor.hpp"
+
+namespace ttnn::operations::moreh::moreh_linear_backward {
+
+void MorehBiasAddBackwardOperation::validate_inputs(
+    const operation_attributes_t& operation_attributes, const tensor_args_t& tensor_args) {
+    auto& bias_grad = tensor_args.bias_grad;
+
+    if (bias_grad.has_value()) {
+        auto bias_grad_shape = bias_grad->get_shape();
+        auto bias_grad_tensor = bias_grad.value();
+        TT_FATAL(
+            tt::operations::primary::is_scalar(bias_grad_tensor) ||
+                tt::operations::primary::is_1d_tensor(bias_grad_tensor),
+            "bias_grad tensor should be 1d or scalar");
+    }
+}
+
+MorehBiasAddBackwardOperation::program_factory_t MorehBiasAddBackwardOperation::select_program_factory(
+    const operation_attributes_t& operation_attributes, const tensor_args_t& tensor_args) {
+    const auto& bias_grad = tensor_args.bias_grad.value();
+    if (tt::operations::primary::is_scalar(bias_grad))
+        return SingleCoreProgramFactory();
+    return MultiCoreProgramFactory();
+}
+
+void MorehBiasAddBackwardOperation::validate_on_program_cache_miss(
+    const operation_attributes_t& operation_attributes, const tensor_args_t& tensor_args) {
+    validate_inputs(operation_attributes, tensor_args);
+};
+
+void MorehBiasAddBackwardOperation::validate_on_program_cache_hit(
+    const operation_attributes_t& operation_attributes, const tensor_args_t& tensor_args) {
+    validate_inputs(operation_attributes, tensor_args);
+};
+
+MorehBiasAddBackwardOperation::shape_return_value_t MorehBiasAddBackwardOperation::compute_output_shapes(
+    const operation_attributes_t& operation_attributes, const tensor_args_t& tensor_args) {
+    auto input_tensor_shape = tensor_args.bias.value().get_shape();
+
+    return input_tensor_shape;
+};
+
+MorehBiasAddBackwardOperation::tensor_return_value_t MorehBiasAddBackwardOperation::create_output_tensors(
+    const operation_attributes_t& operation_attributes, const tensor_args_t& tensor_args) {
+    const auto& output_shape = compute_output_shapes(operation_attributes, tensor_args);
+    auto dtype = tensor_args.bias.value().get_dtype();
+    Layout layout{Layout::TILE};
+    auto device = tensor_args.bias.value().device();
+
+    auto bias_grad_memory_config = operation_attributes.bias_grad_memory_config;
+
+    if (tensor_args.bias_grad.has_value()) {
+        return tensor_args.bias_grad.value();
+    } else {
+        return create_device_tensor(output_shape, dtype, layout, device, bias_grad_memory_config);
+    }
+}
+
+std::tuple<MorehBiasAddBackwardOperation::operation_attributes_t, MorehBiasAddBackwardOperation::tensor_args_t>
+MorehBiasAddBackwardOperation::invoke(
+    const Tensor& output_grad,
+    const std::optional<Tensor>& bias,
+    const std::optional<Tensor>& bias_grad,
+
+    const std::optional<ttnn::MemoryConfig>& bias_grad_memory_config,
+    const ttnn::DeviceComputeKernelConfig ck) {
```

Comment:
Need change this.

Reply:
`compute_kernel_config` can be init by init_device_compute_kernel_config

Reply:
resolved in [3f363ae](https://github.com/tenstorrent/tt-metal/pull/12812/commits/3f363ae0ef41cd058967757ab165f68c13da2d26)